api/v1beta1/machine_types.go

```diff
@@ -97,6 +100,11 @@ type MachineSpec struct {
 	// +optional
 	NodeDrainTimeout *metav1.Duration `json:"nodeDrainTimeout,omitempty"`
```

Comment:
It'd be good to start a section may be here https://cluster-api.sigs.k8s.io/tasks/automated-machine-management/index.html and document the different available timeouts and illustrating examples for their use cases. This can be a follow up though.

Reply:
yes absolutely, I would also prefer to do it in a follow-up PR if there are no objections. Thanks for reminding.